apps/block_scout_web/lib/block_scout_web/controllers/api/v2/internal_transaction_controller.ex

```diff
@@ -29,20 +27,12 @@ defmodule BlockScoutWeb.API.V2.InternalTransactionController do
   """
   @spec internal_transactions(Plug.Conn.t(), map()) :: Plug.Conn.t()
   def internal_transactions(conn, params) do
-    if MigrationStatus.get_status(
-         CreateInternalTransactionsBlockNumberDescTransactionIndexDescIndexDescIndex.migration_name()
-       ) == "completed" do
-      paging_options = paging_options(params)
-
-      options =
-        paging_options
-        |> Keyword.update(:paging_options, default_paging_options(), fn %PagingOptions{
-                                                                          page_size: page_size
-                                                                        } = paging_options ->
-          maybe_parsed_limit = Helper.parse_integer(params["limit"])
-          %PagingOptions{paging_options | page_size: min(page_size, maybe_parsed_limit && abs(maybe_parsed_limit))}
-        end)
-        |> Keyword.merge(@api_true)
+    with true <-
+           BackgroundMigrations.get_heavy_indexes_create_internal_transactions_block_number_desc_transaction_index_desc_index_desc_index_finished(),
+         paging_options = paging_options(params),
+         transaction_hash = transaction_hash_from_params(params),
+         false <- transaction_hash == :invalid do
+      options = options(paging_options, %{transaction_hash: transaction_hash, limit: params["limit"]})
```

Comment:
very much a nitpick: What about placing `paging_options` within the `do` block?

```suggestion
    with true <-
           BackgroundMigrations.get_heavy_indexes_create_internal_transactions_block_number_desc_transaction_index_desc_index_desc_index_finished(),
         transaction_hash when transaction_hash != :invalid <- transaction_hash_from_params(params) do
      paging_options = paging_options(params)
      options = options(paging_options, %{transaction_hash: transaction_hash, limit: params["limit"]})
```